parsons/databases/redshift/rs_create_table.py

```diff
@@ -91,10 +101,8 @@ def data_type(self, val, current_type):
                 if not self.is_valid_integer(val):
                     return 'varchar'
 
-                # Use smallest possible int type
-                if (-32768 < t < 32767) and current_type not in ['int', 'bigint']:
-                    return 'smallint'
-                elif (-2147483648 < t < 2147483647) and current_type not in ['bigint']:
+                # Use smallest possible int type (but don't bother with smallint)
+                if (-2147483648 < t < 2147483647) and current_type not in ['bigint']:
```

Comment:
Is there any chance this would run in an environment where these comparison numbers would cause overflows during execution?

Reply:
So, my understanding is that [Python handles that under the hood](https://mortada.net/can-integer-operations-overflow-in-python.html).